qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -0,0 +1,690 @@
+        """
+        return self._series_curve_fit(curve_data=curve_data, **options)
+
+    # pylint: disable = unused-argument, missing-return-type-doc
+    def _create_figure(self, curve_data: List[CurveEntry], fit_data: AnalysisResult):
+        """Create new figure with the fit result and raw data.
+
+        Subclass can override this method to return figures.
+
+        Args:
+            curve_data: List of raw curve data points with metadata.
+            fit_data: Analysis result containing fit parameters.
+
+        Returns:
+            List of figures (format TBD).
+        """
+        # TODO implement default figure. Will wait for Qiskit-terra #5499
+        return list()
+
+    @staticmethod
+    def _data_processing(data: List[Dict[str, Any]]) -> Tuple[np.ndarray, np.ndarray]:
+        """Format x-values, y-values and y-sigmas from list of circuit result entry.
+
+        .. notes::
+            This method receives full data list of a single curve.
+
+            This is sometime convenient for handling level1 (Kerneled) data.
+            The level1 scatter data may spread in the IQ plane, and it may require
+            full experimental results to calculate the principal component to
+            enhance the signal-to-noise ratio of following analysis.
+
+        Args:
+            data: List of circuit result entry.
+
+        Returns:
+            - y_values: Numpy array of formatted y data values.
+            - y_sigmas: Numpy array of formatted y error values.
+        """
+        try:
+            n_qubits = len(data[0]["metadata"]["qubits"])
+        except KeyError:
+            n_qubits = 1
+
+        # TODO data processor may be initialized with some variables.
+        # For example, if it contains front end discriminator, it may be initialized with
+        # some discrimination line parameters. These parameters cannot be hard-coded here.
+
+        y_values = np.zeros(len(data), dtype=float)
+        y_sigmas = np.zeros(len(data), dtype=float)
+        for idx, datum in enumerate(data):
+            # TODO this should be replaced with preset DataProcessor.
+
+            # TODO fix sigma definition
+            # When the count is 100% zero (i.e. simulator), this yields sigma=0.
+            # This crashes scipy fitter when it calculates covariance matrix (zero-div error).
+
+            y_val, y_err = level2_probability(datum, outcome="1" * n_qubits)
+            y_values[idx] = y_val
+            y_sigmas[idx] = y_err
+
+        return y_values, y_sigmas
+
+    @staticmethod
+    def _post_processing(analysis_result: AnalysisResult) -> AnalysisResult:
+        """Calculate new quantity from the fit result.
+
+        Subclass can override this method to do post analysis.
+
+        Args:
+            analysis_result: Analysis result containing fit result.
+
+        Returns:
+            New AnalysisResult instance containing the result of post analysis.
+        """
+        return analysis_result
```

Comment:
Also here, not sure whether to return a dictionary or an `AnalysisResultV1` object